imports/api/questions.js

```diff
@@ -195,84 +197,56 @@ if (Meteor.isServer) {
-      if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else if (user.isInstructorAnyCourse()) {
-        const courses = _.pluck(Courses.find({instructors: this.userId}).fetch(), '_id')
-        return Questions.find({
-          '$or': [{owner: this.userId}, {courseId: { '$in': courses }, approved: true}],
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else {
-        // students. By checking for creator, they can see the questions they submitted
-        // that have been moved to a course library (which changes the owner w/o copying)
-        return Questions.find({
-          '$or': [{creator: this.userId}, {owner: this.userId}],
-          sessionId: {$exists: false} })
+      const course = Courses.findOne({ _id: courseId})
+      let query = { 
+        sessionId: {$exists: false},
+        courseId: courseId,
       }
-    } else this.ready()
-  })
-
-  Meteor.publish('questions.libraryInCourse', function (courseId) {
-    if (this.userId) {
-      const user = Meteor.users.findOne({_id: this.userId})
       if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false},
-          courseId: courseId
-         })
+        query = _.extend({ studentCopyOfPublic: {$exists: false} }, query)
       } else if (user.isInstructor(courseId)) {
-        return Questions.find({
-          owner: this.userId,
-          courseId: courseId,
-          approved: true,
-          sessionId: {$exists: false}
-        })
+        query = _.extend({ owner: this.userId }, query)
       } else {
         // students. By checking for creator, they can see the questions they submitted
         // that have been moved to a course library (which changes the owner w/o copying)
-        return Questions.find({
+        query = {
           '$or': [{creator: this.userId}, {owner: this.userId}],
           courseId: courseId,
-          sessionId: {$exists: false} })
+          sessionId: {$exists: false} }
       }
+      return Questions.find(query)
     } else this.ready()
   })
 
-  // truly public questions
-  Meteor.publish('questions.public', function () {
-    if (this.userId) {   
-      let cArr = _(Courses.find({ students: this.userId }).fetch()).pluck('_id').concat(_(Courses.find({ instructors: this.userId }).fetch()).pluck('_id'))
-      return Questions.find({ public: true, courseId: {$in: cArr} })
-    } else this.ready()
-  })
-
-  Meteor.publish('questions.publicInCourse', function (courseId) {
+  Meteor.publish('questions.public', function (courseId) {
     if (this.userId) {
-      return Questions.find({ public: true, courseId: courseId })
+      const course = Courses.findOne({ _id: courseId })
```

Comment:
you need to check that userID is either in instructors or in students for that course.